app/spreadsheet.py

```diff
@@ -32,17 +33,19 @@ def add_resource_worksheet_to_workbook(workbook, resources, resource_name):
 
 
 def upload_spreadsheet_to_s3(filename, workbook):
-    # save the workbook to the cwd by closing
+    cwd = os.getcwd()
+    # save the workbook to a temporary directory
+    temp_directory = tempfile.TemporaryDirectory(ignore_cleanup_errors=True)
```

Comment:
Does this imply the file isn't actually created until the `close` command is called? If so, I think you can revert the changes to the Dockerfile

Reply:
Consider [with](https://realpython.com/python-with-statement/) or a try-finally to ensure cleanup

Reply:
Implementing a try-finally now